app/controllers/lead_providers/partnership_csv_uploads_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class PartnershipCsvUploadsController < ::LeadProviders::BaseController
+    def new
+      @partnership_csv_upload = PartnershipCsvUpload.new
+    end
+
+    def create
+      # TODO: do we want to catch empty session[:delivery_partner_id] ?
+      if params[:partnership_csv_upload].blank?
+        @partnership_csv_upload = PartnershipCsvUpload.new
+        @partnership_csv_upload.errors[:base] << "Please select a CSV file to upload."
+        render :new and return
+      end
+
+      @partnership_csv_upload = PartnershipCsvUpload.new(upload_params.merge(
+                                                           lead_provider_id: current_user.lead_provider_profile.lead_provider.id,
+                                                           delivery_partner_id: session[:delivery_partner_id],
+                                                         ))
```

Comment:
I don't fully understand. So the lead provider and delivery partner ids can come from `upload_params`? But we also manually override them here by `merge`?